Update requirements, travis install from .txt

pjsier opened the pull request:
This should address the installation issue you've been seeing in builds. It doesn't happen often, but sometimes I have an issue with `pipenv install` in CI builds so to be safe this is installing from `requirements.txt` in Travis.

Also updates the version of `city-scrapers-core` and other dependencies and includes the `aws` extra for S3 deployments

bonfirefan commented:
Thanks @pjsier, does this mean that we'll need to generate a `requirements.txt` whenever we update our dependencies through pipenv?

pjsier commented:
Was running into some errors with `pitt_city_council` because the tests were patching `parse` instead of `_parse_events` so the spider was actually running against the live site. Updated that and fixed some of the tests so that they pass against the fixture data.

Also removed `scrapy-sentry` because that was causing issues with 3.7. And yeah, you can just run `pipenv lock -r > requirements.txt` and then `pipenv lock -r --dev >> requirements.txt` to get all of the dependencies into that file
